fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -282,7 +280,7 @@ public String getClusterFile() {
      * @return the database factory
      */
     @Nonnull
-    protected FDBDatabaseFactory getFactory() {
+    protected FDBDatabaseFactoryBase getFactory() {
```

Comment:
Unfortunately this is still an API breaking change on a `API.Status.STABLE` class -- all existing users of this method would have to be updated because the return value has changed.  I think you are going to need to reverse these roles, and have `FDBDatabaseFactory` be an `abstract` class, the existing Record Layer version, be an `FDBDatabaseFactoryImpl`.

Reply:
As long as both `setRunLoopProfilingEnabled` and `setThreadsPerClientVersion` are fine to move to FDBDatabaseFactoryImpl that would make sense (otherwise those are clearly relevant to the specific implementation of the native library and not part of the base class).